### ISSUE TITLE
Typescript definitions in NPM are outdated

### ISSUE THREAD

**Issue by OrkhanAlikhanov** [arnog/mathlive#851]
## Description

It seems typescript definitions in NPM are outdated.

Run `npm install mathlive@0.59.0`

in `node_modules/mathlive/dist/public/mathlive.d.ts` file the definition is like this:
```ts
export declare function makeMathField(element: HTMLElement | string, options: MathfieldOptions): Mathfield;
```
It should be generated like so though:
```ts
export declare function makeMathField(element: HTMLElement | string, options?: Partial<MathfieldOptions>): Mathfield;
```

**Comment by OrkhanAlikhanov** [comment]
Also, `makeMathField` is deprecated. Isn't it soon to move to html custom elements? Mathlive may not work in older browsers ini that case, e.g in iOS 9 devices. [See caniuse](https://caniuse.com/custom-elementsv1)

**Comment by arnog** [comment]
This is fixed in main.

**Comment by OrkhanAlikhanov** [comment]
Sorry I did not understand what you meant.

**Comment by arnog** [comment]
@OrkhanAlikhanov When I said "This is fixed in main" I meant that the issue with the type definition of the `options` argument is a problem with 0.59 but is fixed in the main branch in GitHub.

**Comment by OrkhanAlikhanov** [comment]
Ah you mean it's fixed but haven't been published to npm as version `0.60`.

Could you give your thoughts on https://github.com/arnog/mathlive/issues/851#issuecomment-792147053 ?

**Comment by arnog** [comment]
Support for Web Components is currently at 95%. 
iOS 9 was released 5 years ago. The oldest device supporting iOS 9, the iPhone 4s (introduced ten years ago) is no longer supported by its manufacturer since July 22, 2019. Older versions of MathLive will presumably continue to work in these environments.

**Comment by OrkhanAlikhanov** [comment]
Thank you for the clarification. I appreciate

**Comment by OrkhanAlikhanov** [comment]
In not so developed countries, people still rely on older devices. Older android devices, or older versions of windows 10 which had been shipped with legacy microsoft edge (12-18) which never supported this feature. It may be that globally it's 95% coverage, I believe in some countries the value might be around ~60% or so. 


Is there any specific benefit of migrating to web-components or?

**Comment by arnog** [comment]
In general, MathLive supports the last two major versions of evergreen browsers. Unfortunately, supporting older browsers is not something within the reach of an unfunded open source project.

If your project require supporting older devices, you are more than welcome to keep using older versions of MathLive.

The benefits of web components were covered in #665:
* easier integration with UI libraries and frameworks, including reactive libraries
* encapsulation of the DOM elements and stylesheets used by the mathfield: as a web component implemented behind a shadow DOM, the DOM elements and styles of the component are invisible to the DOM outside of the element, and vice-versa, protecting from unwanted interaction between them
* simpler programming model built on top of standard DOM elements conventions, for example using `'content-did-change'` events instead of `onContentDidChange` callbacks, or calling methods directly on the element, i.e. `getElementById('mf').$latex('f(x) = x + 1')`
* declarative syntax to configure a mathfield, for example: `<math-field virtual-keyboard-mode="manual"></math-field>`, while still supporting a programmatic way to create elements (i.e. `new MathfieldElement({virtualKeyboardMode: "manual"})`)
* proper handling of focusing, including properly respecting `tabindex` attribute (see #544)

**Comment by arnog** [comment]
Fixed in mathlive@0.60.0

**Comment by hanzallahmsd** [comment]
How to use this in React? Googling from the past few days but nothing is found sot far.
